sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -238,16 +238,20 @@ class SparkSqlAstBuilder extends AstBuilder {
     ctx.tableProperty.asScala.map { property =>
       // A key can either be a String or a collection of dot separated elements. We need to treat
       // these differently.
-      val key = if (property.key.STRING != null) {
-        string(property.key.STRING)
-      } else {
-        property.key.getText
-      }
+      val key = visitTablePropertyKey(property.key)
       val value = Option(property.value).map(string).orNull
       key -> value
     }.toMap
   }
 
+  override def visitTablePropertyKey(key: TablePropertyKeyContext): String = {
```

Comment:
Could you add a little bit of documentation for this method. The comment on line 239 is a good starting point.